### PR TITLE
Address A11Y comments related to WCAG 2.1

### LINKED ISSUE #519: APA WG comment: Reference to WCAG 2.1
We appreciate the addition of section [D.1 WCAG Considerations](https://www.w3.org/TR/2019/WD-ttml-imsc1.2-20191128/#wcag-applications) and its content.  This is very helpful.  We would, though,
like to change the phrasing "WCAG 2.1 recommends" to "WCAG 2.1 requires", because the WCAG success criteria are normative requirements rather than recommendations.

### PR DESCRIPTION
**Pull request by palemieux**
Closes #519 and #520

***
<a href="https://pr-preview.s3.amazonaws.com/w3c/imsc/pull/526.html" title="Last updated on May 14, 2020, 5:30 PM UTC (2e3298f)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/imsc/526/46949b7...2e3298f.html" title="Last updated on May 14, 2020, 5:30 PM UTC (2e3298f)">Diff</a>

### CODE REVIEW

File: imsc1/spec/ttml-ww-profiles.html [hunk 1]
```diff
@@ -4609,34 +4611,62 @@ <h2>WCAG and MAUR Considerations</h2>
...
+        <h4>Success Criterion 1.1.1 Non-text Content</h4>
+
+        <p><a data-cite="!WCAG21#non-text-content">Success Criterion 1.1.1</a> at [[!WCAG21]] specifies that, with some exceptions,
+        all <a>non-text content</a> that is presented to the user has a <a>text alternative</a> that serves the equivalent
+        purpose.</p>
+
+        <p>In the context of this specification, <a>text alternative</a> is intended primarily to support users of the subtitles who
+        cannot see images (the <a>non-text content</a>). Since the images of an <a>Image Profile</a> <a>Document Instance</a>
+        usually represent subtitle or caption text, the guidelines for authoring text equivalent strings given at <span
+        class='sec-no'>Images of text</span> of [[HTML5]] are appropriate.</p>
```
Comment (nigelmegitt) [inline imsc1/spec/ttml-ww-profiles.html]: I can't find this section in HTML5. Is it somewhere under https://html.spec.whatwg.org/#alt ?

File: imsc1/spec/ttml-ww-profiles.html [hunk 2]
```diff
@@ -4609,34 +4611,62 @@ <h2>WCAG and MAUR Considerations</h2>
...
+
+        <p>Thus, for each subtitle in an <a>Image Profile</a> <a>Document Instance</a>, a text equivalent content in a <a>Text
+        Profile</a> <a>Document Instance</a> SHOULD be written so that it conveys all essential content and fulfills the same
+        function as the corresponding subtitle image. In the context of subtitling and captioning, this content will be (as a
+        minimum) the verbatim equivalent of the image without précis or summarization. However, the author MAY include extra
+        information to the text equivalent string in cases where styling is applied to the text image with a deliberate connotation,
+        as a <em>functional</em> replacement for the applied style.</p>
+
+        <aside class="example">In subtitling and captioning, italics can be used to indicate an off screen speaker context (for
```
Comment (nigelmegitt) [inline imsc1/spec/ttml-ww-profiles.html]: I think this is too suggestive that use of italics in this way is a globally accepted practice.

```suggestion
       <aside class="example">Consider a subtitling and captioning practice that uses italics to indicate an off screen speaker context (for
```

File: imsc1/spec/ttml-ww-profiles.html [hunk 3]
```diff
@@ -4609,34 +4611,62 @@ <h2>WCAG and MAUR Considerations</h2>
...
+        example a voice from a radio). An author can choose to include this functional information in the text equivalent; for
```
Comment (nigelmegitt) [inline imsc1/spec/ttml-ww-profiles.html]: It's not clear here why the author would not use `tts:fontStyle="italic"` here. 

```suggestion
        example a voice from a radio). As an alternative or addition to specifying <code>tts:fontStyle="italic"</code>, an author could choose to include this semantic information in the text equivalent; for
```

File: imsc1/spec/ttml-ww-profiles.html [hunk 4]
```diff
@@ -4609,34 +4611,62 @@ <h2>WCAG and MAUR Considerations</h2>
...
+        example, by including the word "Radio: " before the image equivalent text. Note that images in an <a>Image Profile</a>
+        <a>Document Instance</a> that are intended for use as <em>captions</em>, i.e. intended for a hard of hearing audience, might
+        already include this functional information in the rendered text.</aside>
```
Comment (nigelmegitt) [inline imsc1/spec/ttml-ww-profiles.html]: s/functional information/presentation style ?

File: imsc1/spec/ttml-ww-profiles.html [hunk 5]
```diff
@@ -4609,34 +4611,62 @@ <h2>WCAG and MAUR Considerations</h2>
...
+      </section>
+
+      <section class='appendix' id='wcag-criterion-text=spacing'>
+        <h4>Success Criterion 1.4.12 Text Spacing</h4>
 			
-			<p><a data-cite="!WCAG21#text-spacing">Guideline 1.4.12</a> of [[!WCAG21]] also recommends that no loss of content or functionality occur when spacing of lines, letters, words and paragraphs is modified by a specified amount.</p>
+        <p><a data-cite="!WCAG21#text-spacing">Success Criterion 1.4.12</a> at [[!WCAG21]] specifies that no loss of content or
+          functionality occur when spacing of lines, letters, words and paragraphs is modified by a specified amount.</p>
+
+        <p>In the context of this specification, this is achieved by specifying an appropriate combination of font and text styles,
```
Comment (nigelmegitt) [inline imsc1/spec/ttml-ww-profiles.html]: What is a "font style" here as opposed to a "text style"? I'm not sure if this means attributes applied to a `<font>` element or if it is making a distinction between style (`tts:`) attributes that affect font vs other style attributes?

File: imsc1/spec/ttml-ww-profiles.html [hunk 6]
```diff
@@ -4609,34 +4611,62 @@ <h2>WCAG and MAUR Considerations</h2>
...
+
+      <section class='appendix' id='wcag-criterion-language'>
+        <h4>Success Criteria 3.1.1 Language of Page and 3.1.2 Language of Parts</h4>
+			
+        <p>Success Criteria <a data-cite="!WCAG21#language-of-page">3.1.1</a> and
+        <a data-cite="!WCAG21#language-of-parts"></a>3.1.2</a> at [[!WCAG21]] specify that the human language of each passage or
+        phrase in the content can be programmatically determined.</p>
+
+        <p>In the context of this specification, this is achieved by specifying the appropriate value of <code>xml:lang</code> on
+        the appropriate elements.</p>
```
Comment (nigelmegitt) [inline imsc1/spec/ttml-ww-profiles.html]: Is it worth adding text here that notes that `xml:lang` is inherited so people don't get the idea that it needs to be explicitly set everywhere even if it is being set to the same value?

File: imsc1/spec/ttml-ww-profiles.html (hunk 1 above)
Comment (palemieux) [reply imsc1/spec/ttml-ww-profiles.html]: Perhaps https://html.spec.whatwg.org/#text-that-has-been-rendered-to-a-graphic-for-typographical-effect more specifically?

File: imsc1/spec/ttml-ww-profiles.html (hunk 5 above)
Comment (palemieux) [reply imsc1/spec/ttml-ww-profiles.html]: > if it is making a distinction between style (tts:) attributes that affect font vs other style attributes?

Yes.

File: imsc1/spec/ttml-ww-profiles.html (hunk 3 above)
Comment (palemieux) [reply imsc1/spec/ttml-ww-profiles.html]: This text was the result of a long [thread](https://lists.w3.org/Archives/Public/public-tt/2014Jul/0000.html). I am reluctant to modifying it unless absolutely necessary.

File: imsc1/spec/ttml-ww-profiles.html (hunk 6 above)
Comment (palemieux) [reply imsc1/spec/ttml-ww-profiles.html]: See simplified text.

File: imsc1/spec/ttml-ww-profiles.html (hunk 2 above)
Comment (palemieux) [reply imsc1/spec/ttml-ww-profiles.html]: See revised text.

File: imsc1/spec/ttml-ww-profiles.html (hunk 1 above)
Comment (nigelmegitt) [reply imsc1/spec/ttml-ww-profiles.html:4624]: Looks reasonable to me.

File: imsc1/spec/ttml-ww-profiles.html [hunk 7]
```diff
@@ -4630,11 +4630,11 @@ <h4>Success Criterion 1.1.1 Non-text Content</h4>
...
-        example a voice from a radio). An author can choose to include this functional information in the text equivalent; for
-        example, by including the word "Radio: " before the image equivalent text. Note that images in an <a>Image Profile</a>
-        <a>Document Instance</a> that are intended for use as <em>captions</em>, i.e. intended for a hard of hearing audience, might
-        already include this functional information in the rendered text.</aside>
+        <aside class="example">Considering a subtitling and captioning practice where italics are used to indicate an off screen
+        speaker context (for example a voice from a radio), an author can choose to include this functional information in the text
+        equivalent; for example, by including the word "Radio: " before the image equivalent text. Note that images in an <a>Image
+        Profile</a> <a>Document Instance</a> that are intended for use as <em>captions</em>, i.e. intended for a hard of hearing
+        audience, might already include this functional information in the rendered text.</aside>
```
Comment (nigelmegitt) [inline imsc1/spec/ttml-ww-profiles.html:4637]: I'm not clear on whether "might already include this functional information" is referring to the italic styling or to the added word "Radio: ", and this seems important to this note. I _think_ it is saying that the image might render text in italics but the text equivalent might include "Radio: " and not be in italics. Also should we not include `ttm:role` here?

File: imsc1/spec/ttml-ww-profiles.html (hunk 3 above)
Comment (nigelmegitt) [reply imsc1/spec/ttml-ww-profiles.html]: Thanks for the reminder about that thread. I see that some of the concerns I am raising now relate to the proposal that John made a long time ago. Nevertheless, I think they are valid, and if we are revisiting the text I would like to see if we can address them.

File: imsc1/spec/ttml-ww-profiles.html (hunk 5 above)
Comment (nigelmegitt) [reply imsc1/spec/ttml-ww-profiles.html:4655]: > Yes.

Thanks, in that case:

```suggestion
        <p>In the context of this specification, this is achieved by applying an appropriate combination of styles to layout and content elements to achieve the desired
```

File: imsc1/spec/ttml-ww-profiles.html (hunk 6 above)
Comment (nigelmegitt) [reply imsc1/spec/ttml-ww-profiles.html]: Less is more, eh? OK!

File: imsc1/spec/ttml-ww-profiles.html (hunk 2 above)
Comment (nigelmegitt) [reply imsc1/spec/ttml-ww-profiles.html]: Thanks, that works for me.

**Comment by nigelmegitt** [comment]
@gzimmermann alerting you to this pull request. I can't add you as a formal reviewer but it would be helpful to know you've seen it and if it addresses the issue for you.

File: imsc1/spec/ttml-ww-profiles.html (hunk 3 above)
Comment (nigelmegitt) [reply imsc1/spec/ttml-ww-profiles.html]: I've raised #551 so we can make progress with this PR and handle this comment separately.

File: imsc1/spec/ttml-ww-profiles.html (hunk 4 above)
Comment (nigelmegitt) [reply imsc1/spec/ttml-ww-profiles.html]: I've raised #551 so we can make progress with this PR and handle this comment separately.

File: imsc1/spec/ttml-ww-profiles.html (hunk 7 above)
Comment (nigelmegitt) [reply imsc1/spec/ttml-ww-profiles.html:4638]: I've raised #551 so we can make progress with this PR and handle this comment separately.

**Comment by nigelmegitt** [comment]
I decided not to request any further changes here in this pull request and moved my comments out to a new issue at #551 .

**Comment by nigelmegitt** [comment]
As per https://github.com/w3c/imsc/issues/519#issuecomment-628753834 this pulll request addresses the concerns raised in #519.

**Comment by nigelmegitt** [comment]
As per https://github.com/w3c/imsc/issues/520#issuecomment-628796066 this pull request addresses the concerns raised in #520.

**Pull request merged**